File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops)

(defn convert [n] ;; <- number
      ;; your code goes here
      (let [
            d3 (zero? (mod n 3))
            d5 (zero? (mod n 5))
            d7 (zero? (mod n 7))
            m {
               "Pling" d3,
               "Plang" d5,
               "Plong" d7
              }
            filtered (clojure.string/join (filter #(get m %) (keys m)))
            ]
        (if (empty? filtered)
          (str n)
          filtered)
        )
)

File: solutions/python/raindrops/1/raindrops.py
def convert(number):
    div_string_pairs = {"Pling": 3, "Plang": 5, "Plong": 7}
    s = ""
    for string, divisor in div_string_pairs.items():
        if number % divisor == 0:
            s += string
    if s:
        return s
    return str(number)
